js/interface.js

```diff
@@ -79,38 +77,50 @@ $('#show-full-name-field').on('click', function() {
 
 // Fired from Fliplet Studio when the external save button is clicked
 Fliplet.Widget.onSaveRequest(function () {
-  $('form').submit();
+  dataSourceProvider.forwardSaveRequest();
 });
 
-$dataSources.on('change', function() {
-  var selectedDataSourceId = $(this).val();
-  if (selectedDataSourceId === 'none') {
-    $('#manage-data').addClass('hidden');
-    $('.column-selection').removeClass('show');
-    return;
-  }
-
-  if (selectedDataSourceId === 'new') {
-    $('#manage-data').addClass('hidden');
-    createDataSource();
-    return;
-  }
+function initDataSourceProvider(currentDataSourceId) {
+  var dataSourceData = {
+    dataSourceTitle: 'Chat data source',
+    dataSourceId: currentDataSourceId,
+    appId: Fliplet.Env.get('appId'),
+    default: {
+      name: 'Chat data for' + Fliplet.Env.get('appName'),
+      entries: [],
+      columns: []
+    },
+    accessRules: [
+      {
+        allow: 'all',
+        enabled: true,
+        type: [
+          'select'
```

Comment:
I believe chat also needs to be able to update the data source records.

Reply:
@squallstar Should chat only update and ready record or adding them as well?

Reply:
just read and update

Reply:
@squallstar Added